spras/analysis/summary.py

```diff
@@ -50,8 +51,44 @@ def summarize_networks(file_paths: Iterable[Path], node_table: pd.DataFrame, alg
         number_edges = nw.number_of_edges()
         ncc = nx.number_connected_components(nw)
 
+        # Save the max/median degree, average clustering coeffienct, and density
+        if number_nodes <= 1 or number_edges == 0:
+            max_degree = 0
+            median_degree = 0.0
+            density = 0.0
+        else:
+            degrees = [deg for _, deg in nw.degree()]
+            max_degree = max(degrees)
+            median_degree = median(degrees)
+            density = nx.density(nw)
+
+        # Save the max diameter
+        cc = list(nx.connected_components(nw))
+        diameters = [
+            nx.diameter(nw.subgraph(c).copy()) if len(c) > 1 else 0
```

Comment:
The restated comment here ("Save the max diameter") is less helpful than adding a comment about why we ignore the output of `nx.diameter` when the there is less than 2 nodes in any one individual component nodeset.